run_bot_nix.sh

```
@@ -69,20 +83,42 @@ create_output_dir() {
 # Process JSON configuration to add UUID if missing
 process_config() {
     local config_json=$1
+    local use_api_mode=$2
     local bot_uuid=$(generate_uuid)
     
     print_info "Generated new bot_uuid: $bot_uuid" >&2
+
+    # Load environment variables from .env file only if needed for API mode
+    if [ "$use_api_mode" = "true" ]; then
+        if [ -f ".env" ]; then
+            source .env
```

Shellcheck (0.10.0)

[info] Not following: .env was not specified as input (see shellcheck -x).

(SC1091)